libnmstate/state.py

```diff
@@ -939,3 +945,53 @@ def _remove_route_rule_default_values(rule):
     if rule.get(RouteRule.ROUTE_TABLE) == RouteRule.USE_DEFAULT_ROUTE_TABLE:
         del rule[RouteRule.ROUTE_TABLE]
     return rule
+
+
+def merge_linux_bridge_ports(desired_iface_state, current_iface_state):
+    """
+    Given a linux bridge desired state, and it's current state, merges
+    those together.
+
+    This extension of the interface merging mechanism simplifies the user's
+    life in scenarios where the user wants to partially update the bridge's
+    configuration - e.g. update only the bridge's port STP configuration -
+    since it enables the user to simply specify the updated values rather
+    than the full current state + the updated value.
+    """
+    desired_bridge_state = desired_iface_state.get(
+        LinuxBridge.CONFIG_SUBTREE, {}
+    )
+    desired_ports = desired_bridge_state.get(LinuxBridge.PORT_SUBTREE)
+    current_ports = current_iface_state.get(
+        LinuxBridge.CONFIG_SUBTREE, {}
+    ).get(LinuxBridge.PORT_SUBTREE, [])
+
+    if desired_ports is None and current_ports:
+        desired_iface_state[LinuxBridge.CONFIG_SUBTREE].update(
+            {LinuxBridge.PORT_SUBTREE: current_ports}
+        )
+        return
+    elif not desired_ports:
+        return
+
+    other_index_ports = _index_ports(current_ports)
+    desired_indexed_ports = _index_ports(desired_ports)
+
+    merged_ports = {}
+    for name in desired_indexed_ports.keys() & other_index_ports.keys():
+        dict_update(other_index_ports[name], desired_indexed_ports[name])
+        merged_ports[name] = other_index_ports[name]
+
+    desired_bridge_state[LinuxBridge.PORT_SUBTREE] = list(
+        merged_ports.values()
+    )
+
+
+def _index_ports(ports):
+    indexed_ports = {}
+    for port in ports:
+        port_name = port.get(LinuxBridge.Port.NAME)
+
+        if port_name:
+            indexed_ports[port_name] = port
+    return indexed_ports
```

Comment:
it must have a port name, so just following the interfaces indexing, this should work:
```
        return {
            port[LinuxBridge.Port.NAME]: port for port in ports
        }
```